core/src/main/scala/org/apache/spark/storage/DiskStore.scala

```diff
@@ -301,7 +301,10 @@ private class ReadableChannelFileRegion(source: ReadableByteChannel, blockSize:
     written
   }
 
-  override def deallocate(): Unit = source.close()
+  override def deallocate() {
+    source.close()
+    StorageUtils.dispose(buffer)
```

Comment:
IIRC direct byte buffer can be GCed, cc @kiszk

Reply:
In fact, direct memory is also off-heap memory, 
if we config this parameter "-XX:+DisableExplicitGC",the memory will not be reclaimed

Reply:
Sorry for being late since I overlooked.

While `ByteBuffer.allocateDirect` allocates memory in off-heap memory, the memory will be also reclaimed when the ByteBuffer object will be reclaimed.
Regarding `-XX:+DisableExplicitGC`, as described [here](http://www.oracle.com/technetwork/java/javase/tech/vmoptions-jsp-140102.html), this option just disables `System.gc()`. Other GC operations may reclaim the memory in off-heap allocated by `ByteBuffer.allocateDirect`.
>By default calls to System.gc() are enabled (-XX:-DisableExplicitGC). Use -XX:+DisableExplicitGC to disable calls to System.gc(). Note that the JVM still performs garbage collection when necessary.

Reply:
In other words, I think that the off-heap in `buffer` will not be leaked if `buffer` will be reclaimed.

Reply:
I see,thanks